x/gamm/pool-models/balancer/pool_test.go

```diff
@@ -354,6 +351,42 @@ func TestCalcJoinSingleAssetTokensIn(t *testing.T) {
 			tokensIn:     sdk.NewCoins(sdk.NewInt64Coin("uosmo", 50_000), sdk.NewInt64Coin("uatom", 100_000)),
 			expectShares: sdk.NewInt(2_072_912_400_000_000 + 1_624_999_900_000),
 		},
+		{
+			// Compares expected shares and actual shares to make sure the multiplicativeTolerance ratio below is maintained
+			// |a - b| / min(a, b) <= allowedErrRatio
+			// 0.04000000047920000 = abs(2_599_999_968_750-2_499_999_968_800)/2_499_999_968_800
```

Comment:
Identifying the test cases is an important step toward fixing these issues. You've done a great job at this! Now, we should understand the sources of these errors so that we can fix them.

On that note, I'm wondering: 

- How did you choose these values so that they are outside of the error tolerance? In other words, for each test case, what choice of inputs causes the delta to be large? 

Could you please document this directly in the comments? I think it would be incredibly helpful so that we can come back to it later and know what causes these issues.